rpcserver.go

```diff
@@ -2328,6 +2332,83 @@ func (r *rpcServer) LookupInvoice(ctx context.Context,
 	return rpcInvoice, nil
 }
 
+// createRPCInvoice creates an *lnrpc.Payment from the *channeldb.Payment.
+func createRPCPayment(payment *channeldb.OutgoingPayment) (*lnrpc.Payment, error) {
+
+	paymentHash := sha256.Sum256(payment.PaymentPreimage[:])
+	paymentHashString := hex.EncodeToString(paymentHash[:])
+
+	fee := payment.Fee
+
+	paymentPreImage := hex.EncodeToString(payment.PaymentPreimage[:])
+
+	path := make([]string, len(payment.Path))
+	for i, hop := range payment.Path {
+		path[i] = hex.EncodeToString(hop[:])
+	}
+
+	satAmt := payment.Terms.Value.ToSatoshis()
+
+	return &lnrpc.Payment{
+		PaymentHash:     paymentHashString,
+		Value:           int64(satAmt),
+		CreationDate:    payment.CreationDate.Unix(),
+		Path:            path,
+		Fee:             int64(fee),
+		PaymentPreimage: paymentPreImage,
+	}, nil
+}
+
+// LookupPayment attempts to look up an payment according to its payment hash.
+// The passed payment hash *must* be exactly 32 bytes, if not an error is
+// returned.
+func (r *rpcServer) LookupPayment(ctx context.Context,
+	req *lnrpc.PaymentHash) (*lnrpc.Payment, error) {
+
+	var (
+		payHash [32]byte
+		rHash   []byte
+		err     error
+	)
+
+	// If the RHash as a raw string was provided, then decode that and use
+	// that directly. Otherwise, we use the raw bytes provided.
+	if req.RHashStr != "" {
+		rHash, err = hex.DecodeString(req.RHashStr)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		rHash = req.RHash
+	}
+
+	// Ensure that the payment hash is *exactly* 32-bytes.
+	if len(rHash) != 0 && len(rHash) != 32 {
```

Comment:
When is `rHash` 0 bytes long?

Reply:
I believe it is not, I removed it. Good noticing.